pallets/msa/src/lib.rs

```diff
@@ -1246,14 +1237,42 @@ impl<T: Config + Send + Sync> CheckFreeExtrinsicUse<T> {
 			.into();
 		return ValidTransaction::with_tag_prefix(TAG_PREFIX).and_provides(account_id).build()
 	}
+
+	/// validates that a MSA being retired is valid
+	pub fn validate_msa_retirement(account_id: &T::AccountId) -> TransactionValidity {
```

Comment:
to continue the pattern of using ensure, perhaps, call it 
```
ensure_msa_can_retire
```

Reply:
Sure